tools/toolchain/clang_lto_pgo.sh

```diff
@@ -0,0 +1,108 @@
+#!/bin/bash -uex
+
+if [ -z "${CLANG_BUILD}" ]; then
+	echo "Skip building optimized clang"
+	exit 0
+elif [ "${CLANG_BUILD}" = "SKIP" ]; then
+        echo "Skip building optimized clang"
+        exit 0
+elif [ "${CLANG_BUILD}" = "BUILD" ]; then
+	echo "build optimized clang"
+elif [ "${CLANG_BUILD}" = "INSTALL" ]; then
+	echo "build and install optimized clang"
+else
+	echo "Not sure what to do with ${CLANG_BUILD}"
+	exit 1
+fi
+# Which Scylla branch to train on
+SCYLLA_BRANCH=master
+
+# Which LLVM release to build
+LLVM_TAG=15.0.7
+
+if [ -d "/optimized_clang" ];
+then
+	DIR="/optimized_clang"
+else
+	DIR="${PWD}/optimized_clang"
+	mkdir -p "${DIR}"
+fi
+cd "${DIR}"
+
+# Download the scylla codebase for training.
+# Scylla's dependencies are already installed.
+git clone https://github.com/scylladb/scylla --branch ${SCYLLA_BRANCH} --depth=1
+git -C scylla submodule update --init --depth=1
+
+# Clone, patch and bootstrap the newest Clang and BOLT.
+git clone https://github.com/llvm/llvm-project --branch main --depth=1 stage-0
```

Comment:
Need to pin to a specific version, to get reproducible builds.

Reply:
Agreed. I think there was some clang bug that was not fixed in 15.x, but I'll check if we can use 16 when it GA's.